### ISSUE TITLE
Typescript compiler prevents tree shaking by removing comments. 

### ISSUE THREAD

**Issue by seivan** [Microsoft/TypeScript#21087]
**TypeScript Version:**  2.7.0-dev.201xxxxx

**Code**
https://www.typescriptlang.org/play/index.html#src=class%20Sample%20%7B%0D%0A%20%20%20%20demo%20%3D%20()%20%3D%3E%20%7B%0D%0A%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Aconst%20Button%20%3D%20%2F*%40__PURE__*%2F%20new%20Sample().demo

**Expected behavior:**
The comment should not be stripped 

**Actual behavior:**
The comment is stripped and tree shaking breaks in production.

**Related:**
"Tree shaking" works when using `ES6` modules and targeting the file directly, e.g   

`import { Button } from "lib/Button"`

But if you want, what I want, the glorious way of doing 

`import { Button } from "lib"` 

Then it's no good. UNLESS the transpiled code (the output before minifying) looks like 

`const Button = /*@__PURE__*/ whateverFunctionThatCreatesButton {    }`

In other words, it's a no go with Typescripts compiler as it will remove that comment.

**Comment by weswigham** [comment]
Duplicates https://github.com/Microsoft/TypeScript/issues/17606 (but it is worth remembering that `new` expressions should be tested for, too)

**Comment by seivan** [comment]
@weswigham Should this be closed since its a dupe? 
From my point of view, this is a *decent* sized issue since it prevents tree shaking to work, thus creates larger than necessary bundles that need workarounds to prevent.
